spec/system/avo/location_tag_field_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'LocationTagField', type: :system do
+  describe 'without value' do
+    let(:city) { create :city, coordinates: [] }
+
+    context 'show' do
+      it 'shows empty location field' do
+        visit "/admin/resources/cities/#{city.id}"
+
+        expect(find_field_element('coordinates')).to have_text empty_dash
+      end
+    end
+
+    context 'edit' do
+      it "has lat/long fields and placeholders" do
+        visit "/admin/resources/cities/#{city.id}/edit"
+
+        coordinates_element = find_field_element("coordinates")
+
+        expect(coordinates_element).to have_text "COORDINATES"
+
+        expect(find_by_id("city_coordinates[latitude]", visible: false)).to have_text("")
+        expect(find_by_id("city_coordinates[latitude]", visible: false)[:placeholder]).to have_text("Enter latitude")
+
+        expect(find_by_id("city_coordinates[longitude]", visible: false)).to have_text("")
+        expect(find_by_id("city_coordinates[longitude]", visible: false)[:placeholder]).to have_text("Enter longitude")
+      end
+
+      it "changes the city coordinates" do
+        visit "/admin/resources/cities/#{city.id}/edit"
+
+        fill_in "Enter latitude", with: "1.0"
+        fill_in "Enter longitude", with: "2.0"
+
+        click_on "Save"
+
+        wait_for_loaded
+
+        click_on "Edit"
+
+        expect(find_by_id("city_coordinates[latitude]").value).to eq("1.0")
+        expect(find_by_id("city_coordinates[longitude]").value).to eq("2.0")
+      end
+    end
+  end
+
+  describe 'with regular value' do
+    # Eiffel Tower coordinates used
+    let!(:city) { create :city, latitude: 48.8584, longitude: 2.2945 }
+
+    context "show" do
+      it "renders a map" do
+        visit "/admin/resources/cities/#{city.id}"
+
+        expect(page).to have_css("canvas.mapboxgl-canvas")
```

Comment:
I examined this test to see why it was failing, and I think there may be several things going on here:

1. Headless chrome (which is where these tests are normally running in CI) appears to have issues with using WebGL. I don't have any great links with simple solutions, but you can read a little about [issues with getting webgl to run in headless](https://stackoverflow.com/questions/70948512/how-to-enable-webgl-in-headless-chrome-96-within-selenium-docker-project-to-r). I was able to replicate the failure locally on MacOS Ventura.
2. I tried running it locally in non-headless mode (HEADFULL=true bundle exec rspec spec/system/avo/location_tag_field_spec.rb:58) and while it still failed, the screenshot was different! Instead of the usual Failed to initialize WebGL message, the error shown was An API access token is required to use Mapbox GL. See https://www.mapbox.com/api-documentation/#access-tokens-and-token-scopes. That's very encouraging, because it looks like mapbox did attempt to start in that test run.

Avo currently includes webmock, so that seems to be the convention for mocking out external service dependencies.

Reply:
I'll look into mocking out the Mapbox API for the spec. Thank you for the review and the testing!